electron/api/index.ts

```diff
@@ -5,4 +5,18 @@ export default {
   ...fileSave,
   setUnsavedChanges: (value: boolean) =>
     ipcRenderer.send("set-unsaved-changes", value),
+  send: (channel:string, data:string) => {
+      // whitelist channels
+      const validChannels = ['backend'];
+      if (validChannels.includes(channel)) {
+        ipcRenderer.send(channel, data);
+      }
+    },
+    receive: (channel:string, func:(...arg:string[]) => void) => {
+      const validChannels = ['backend'];
+      if (validChannels.includes(channel)) {
+        // Deliberately strip event as it includes `sender`
+        ipcRenderer.on(channel, (event, ...args:string[]) => func(...args));
+      }
+    }
```

Comment:
Why do this + defining `validChannels` when you could just wrap the channels like with `setUnsavedChanges`?